ng-dev/auth/shared/node-handler.ts

```diff
@@ -0,0 +1,144 @@
+      const error = searchParams.get('error');
+
+      if (!state && !code && !error) {
+        // ignore irrelevant requests (e.g. favicon.ico)
+        return;
+      }
+
+      log('Handling Authorization Request ', searchParams, state, code, error);
+      let authorizationResponse: AuthorizationResponse | null = null;
+      let authorizationError: AuthorizationError | null = null;
+      if (error) {
+        log('error');
+        // get additional optional info.
+        const errorUri = searchParams.get('error_uri') || undefined;
+        const errorDescription = searchParams.get('error_description') || undefined;
+        authorizationError = new AuthorizationError({
+          error: error,
+          error_description: errorDescription,
+          error_uri: errorUri,
+          state: state,
+        });
+      } else {
+        authorizationResponse = new AuthorizationResponse({code: code!, state: state!});
+      }
+      const completeResponse = {
+        request,
+        response: authorizationResponse,
+        error: authorizationError,
+      } as AuthorizationRequestResponse;
+      emitter.emit(ServerEventsEmitter.ON_AUTHORIZATION_RESPONSE, completeResponse);
+      response.setHeader('Content-Type', 'text/html');
+      response.end(this.authorizationRedirectPageContent);
+    };
+
+    this.authorizationPromise = new Promise<AuthorizationRequestResponse>((resolve, reject) => {
+      emitter.once(ServerEventsEmitter.ON_UNABLE_TO_START, () => {
+        reject(`Unable to create HTTP server at port ${this.httpServerPort}`);
+      });
+      emitter.once(ServerEventsEmitter.ON_AUTHORIZATION_RESPONSE, (result: any) => {
+        // Set timeout for the server connections to 1 ms as we wish to close and end the server
+        // as soon as possible. This prevents a user failing to close the redirect window from
+        // causing a hanging process due to the server.
+        server.setTimeout(0);
+        server.close();
+        // resolve pending promise
+        resolve(result as AuthorizationRequestResponse);
+        // complete authorization flow
+        this.completeAuthorizationRequestIfPossible();
+      });
+    });
+
+    let server: Http.Server;
+    request
+      .setupCodeVerifier()
+      .then(() => {
+        server = Http.createServer(requestHandler);
+        server.listen(this.httpServerPort);
+        server.on('error', (e: any) => {
+          if (e.code === 'EADDRINUSE') {
+            console.log('Address in use, retrying...');
+            setTimeout(() => {
+              server.close();
+              server.listen(this.httpServerPort);
+            }, 1000);
+          }
+        });
+        const url = this.buildRequestUrl(configuration, request);
+        if (process.env.DISPLAY) {
+          opener(url);
+        } else {
+          log(`Please visit: ${url}`);
+        }
+      })
+      .catch((error) => {
+        log('Something bad happened ', error);
```

Comment:
```suggestion
        log('An unexpected error occurred', error);
```

(also as mentioned above: why not use dev-infra logging here?)